src/main/java/com/twilio/jwt/validation/ValidationToken.java

```diff
@@ -124,8 +124,14 @@ public static ValidationToken fromHttpRequest(
         builder.headers(request.getAllHeaders());
         builder.signedHeaders(signedHeaders);
 
-        if (HttpMethod.POST.toString().equals(method.toUpperCase())) {
-            HttpEntity entity = ((HttpEntityEnclosingRequest)request).getEntity();
+        /**
+         * If the request encloses an "entity", use it for the body. Note that this is dependent on several factors
+         * during request building and is not solely based on the specified method.
+         *
+         * @see org.apache.http.client.methods.RequestBuilder#build
+         */
+        if (request instanceof HttpEntityEnclosingRequest) {
```

Comment:
Will this cause unintended side effects? Did you consider combining the if statements so `if (POST && enclosing_request)`?

Reply:
Yes, it only affects requests when using the `ValidationClient` (at least internally; any consumer *could* use the `ValdiationToken/Interceptor` classes wherever they want).

I can't think of any negative side-effects nor could I find any internal to the lib. The request builder is [here](https://tinyurl.com/y7okkwk6). This change would fix a bug in the case of PUT requests that contain params/entity.